R/pkg/R/SQLContext.R

```diff
@@ -189,19 +238,67 @@ createDataFrame <- function(data, schema = NULL, samplingRatio = 1.0,
           x
         }
       }
+      data[] <- lapply(data, cleanCols)
 
-      # drop factors and wrap lists
-      data <- setNames(lapply(data, cleanCols), NULL)
+      args <- list(FUN = list, SIMPLIFY = FALSE, USE.NAMES = FALSE)
+      if (arrowEnabled) {
+        shouldUseArrow <- tryCatch({
+          stopifnot(length(data) > 0)
+          dataHead <- head(data, 1)
+          # Currenty Arrow optimization does not support POSIXct and raw for now.
+          # Also, it does not support explicit float type set by users. It leads to
+          # incorrect conversion. We will fall back to the path without Arrow optimization.
+          if (any(sapply(dataHead, function(x) is(x, "POSIXct")))) {
+            stop("Arrow optimization with R DataFrame does not support POSIXct type yet.")
+          }
+          if (any(sapply(dataHead, is.raw))) {
+            stop("Arrow optimization with R DataFrame does not support raw type yet.")
+          }
+          if (inherits(schema, "structType")) {
+            if (any(sapply(schema$fields(), function(x) x$dataType.toString() == "FloatType"))) {
+              stop("Arrow optimization with R DataFrame does not support FloatType type yet.")
```

Comment:
Any idea what's going on with the `FloatType`? Is it a problem on the arrow side?

Reply:
I suspect that it happens when `numeric` (which is like `1.0`) is casted into float type. I think it's related with casting behaviour. Let me take a look and file a JIRA there in Arrow side but if you don't mind I will focus on matching exact type cases for now ...

Reply:
maybe out of bit range? 53 bit https://stat.ethz.ch/R-manual/R-patched/library/base/html/double.html

Reply:
I think it's a bug because it always produces a corrupt value when I try to use `number` as explicit float types.